lib/statsd.js

```diff
@@ -154,6 +154,45 @@ const Client = function (host, port, prefix, suffix, globalize, cacheDns, mock,
     global.statsd = this;
   }
 
+  // only for uds (options.protocol uds)
+  // enabled with the extra flag options.udsGracefulErrorHandling
+  // will gracefully (attempt) to re-open the socket with a small delay
+  // options.udsGracefulRestartRateLimit is the minimum time (ms) between creating sockets
+  // does not support options.isChild (how to re-create a socket you didn't create?)
+  if (!options.isChild && options.protocol === 'uds' && options.udsGracefulErrorHandling) {
+    const socketCreateLimit = options.udsGracefulRestartRateLimit || 1000; // only recreate once per second
```

Comment:
Could you make 1000 a constant at the top of the file?  We don't do this everywhere but trying to get better with it for cases like this.